File: server/Resseract/resseract-server/src/main/java/abistech/resseract/analysis/AnalysisSpecification.java
package abistech.resseract.analysis;

import abistech.resseract.config.Config;
import abistech.resseract.data.frame.DataKey;
import lombok.*;

@Setter
@Getter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class AnalysisSpecification {

    private String analysisName;
    private AnalysisType analysisType;
    private DataKey dataKey;
    private Config configurations;
}

File: angular-app/Resseract/src/app/specs/analysis-specification.ts
import { Configration } from "./analysis-configration";
import { DataKey } from "./data-key";

export class AnalysisSpecification {
    public analysisName: string;
        
    constructor(public analysisType: string,
        public dataKey: DataKey,
        public configurations: Configration) {
    }
}